<launch>
  <!-- <arg name="align_depth"       default="true"/>
  <arg name="unite_imu_method" default="linear_interpolation" />
  <arg name="enable_gyro" default="true"/>
  <arg name="enable_accel" default="true"/>
  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
      <arg name="align_depth" value="$(arg align_depth)"/>
      <arg name="unite_imu_method" value="$(unite_imu_method)"/>
      <arg name="enable_gyro" value="$(enable_gyro)"/>
      <arg name="enable_accel" value="$(enable_accel)"/>
  </include> -->
  <group ns="rtabmap">
  <!-- RGB-D Odometry -->
    <node pkg="nodelet" type="nodelet" name="rgbd_sync" args="standalone rtabmap_sync/rgbd_sync" output="screen">
      <remap from="rgb/image"       to="/camera/color/image_raw"/>
      <remap from="depth/image"     to="/camera/aligned_depth_to_color/image_raw"/>
      <remap from="rgb/camera_info" to="/camera/color/camera_info"/>
      <remap from="rgbd_image"      to="rgbd_image"/> <!-- output -->
      <!-- Should be true for not synchronized camera topics 
           (e.g., false for kinectv2, zed, realsense, true for xtion, kinect360)-->
      <param name="approx_sync"       value="false"/> 
    </node>
 <!-- Odometry -->
    <!-- <node pkg="rtabmap_odom" type="rgbd_odometry" name="rgbd_odometry" output="screen">
      <param name="subscribe_rgbd" type="bool"   value="true"/>
      <param name="odom_frame_id" type="string" value="base_link"/> 
      <param name="frame_id"       type="string" value="camera_link"/>
      <remap from="rgbd_image" to="rgbd_image"/>
    </node> -->
  <!-- RTABMAP SLAM -->
    <node name="rtabmap" pkg="rtabmap_slam" type="rtabmap" output="screen" args="--delete_db_on_start">
          <param name="frame_id" type="string" value="camera_link"/>
          <param name="subscribe_depth" type="bool" value="false"/>
          <param name="subscribe_rgbd" type="bool" value="true"/>

          <!-- <remap from="odom" to="/odometry/filtered"/> -->
          <remap from="odom" to="/odometry/">
          <!-- <remap from="odom" to="/odom"/> -->
          <remap from="rgbd_image" to="rgbd_image"/>
          <param name="queue_size" type="int" value="20"/>

          <!-- RTAB-Map's parameters -->
          <param name="RGBD/AngularUpdate"        type="string" value="0.01"/>
          <param name="RGBD/LinearUpdate"         type="string" value="0.01"/>
          <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>

    </node>
    
    <!-- Visualisation RTAB-Map -->
    <node  pkg="rtabmap_viz" type="rtabmap_viz" name="rtabmap_viz" args="--delete_db_on_start" output="screen" >
      <param name="frame_id"   type="string" value="camera_link"/>
    </node>
  </group>

  <node  pkg="rviz" type="rviz" name="rviz"  output="screen"/>
  
</launch>
